Content.Shared/Gravity/SharedGravitySystem.cs

```diff
@@ -22,18 +23,28 @@ public bool IsWeightless(EntityUid uid, PhysicsComponent? body = null, Transform
             Resolve(uid, ref body, false);
 
             if ((body?.BodyType & (BodyType.Static | BodyType.Kinematic)) != 0)
+            {
+                _appearance.SetData(uid, GravityVisuals.Enabled, false);
                 return false;
+            }
```

Comment:
"IsX" functions should not be modifying state.

Reply:
done